command/ssh/certificate.go

```diff
@@ -403,9 +404,17 @@ func sshAddKeyToAgent(subject string, cert *ssh.Certificate, priv interface{}) e
 		return errors.Wrap(err, "error connecting with ssh-agent")
 	}
 	client := agent.NewClient(conn)
+
+	// ValidAfter and ValidBefore are in ms and the duration expected by the
+	// ssh agent is sec. So we mod by 1000 (removing the first 3 0's) and then
+	// round up. Add this to the difference between ValidAfter and Now, in case
+	// the certificate is scheduled for the future.
+	lifetime := (int64(cert.ValidBefore)-time.Now().UTC().Unix())%1000 + 1
```

Comment:
This does not take into account when the validity is "forever", currently we don't support this, but we might want to take this into account in case we want to support it at some point.

See flag `-V validity_interval` in `man ssh-kegen` for the multiple options.

Reply:
Both ValidAfter and ValidBefore are optional.